src/JsonApi/CompoundDocument.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace Choredo\JsonApi;
+
+class CompoundDocument implements JsonApiResource
+{
+    /**
+     * @var JsonApiResource
+     */
+    private $resource;
+
+    /**
+     * @var array
+     */
+    private $included;
+
+    /**
+     * CompoundDocument constructor.
+     * @param JsonApiResource $resource
+     * @param array $includedData
+     */
+    public function __construct(JsonApiResource $resource, array $includedData)
+    {
+        $this->resource = $resource;
+        $this->included = $includedData;
+    }
+
+    public function getRelatedResource(Relation $relation): ?JsonApiResource
+    {
+        $relationship = array_pop(array_filter($this->included, function (JsonApiResource $included) use ($relation) {
```

Comment:
This is a little hard to read and isn't PSR-2 ✌️

Reply:
3psr5u